sqlmesh/core/snapshot/evaluator.py

```diff
@@ -270,6 +271,7 @@ def create(
             target_snapshots: Target snapshots.
             snapshots: Mapping of snapshot ID to snapshot.
             deployability_index: Determines snapshots that are deployable in the context of this creation.
+            on_start: A callback to initialize the snapshot creation progress bar.
```

Comment:
[Nit] The creation progress bar is just a specific use case and this is a generic callback. I'd rather describe what input arguments for the callback are.